### ISSUE TITLE
Encoding the unit "unitsml(kg*m^(p)*s^(q))" causes parsing error

### ISSUE THREAD

**Issue by ronaldtse** [plurimath/asciimath2unitsml#32]
Reported in https://github.com/metanorma/bipm-si-brochure/issues/174

This works:
```
"unitsml(kg*m*s^(-2)*A^(-2))"
```

But this crashes:
```
"unitsml(kg*m^(p)*s^(q))"
```

The only differences is that the powers are replaced with p and q as variables.

<img width="667" alt="Screenshot 2022-06-21 at 5 18 15 PM" src="https://user-images.githubusercontent.com/11865/174751658-e8461e94-adb3-448e-a664-f88014b8f464.png">

Trace:
```
Fatal Error: 
in source:1 at 3
kg*m^(p)*s^(q)
   ^
bundler: failed to load command: metanorma (/Users/me/.rbenv/versions/3.1.1/bin/metanorma)
/Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/rsec-1.0.0/lib/rsec/parser.rb:19:in `parse!':  (Rsec::SyntaxError)
in source:1 at 3
kg*m^(p)*s^(q)
   ^
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/asciimath2unitsml-0.4.3/lib/asciimath2unitsml/parse.rb:56:in `parse_units'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/asciimath2unitsml-0.4.3/lib/asciimath2unitsml/parse.rb:51:in `parse'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/asciimath2unitsml-0.4.3/lib/asciimath2unitsml/parse.rb:129:in `block in MathML2UnitsML'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/nokogiri-1.13.6-arm64-darwin/lib/nokogiri/xml/node_set.rb:234:in `block in each'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/nokogiri-1.13.6-arm64-darwin/lib/nokogiri/xml/node_set.rb:233:in `upto'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/nokogiri-1.13.6-arm64-darwin/lib/nokogiri/xml/node_set.rb:233:in `each'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/asciimath2unitsml-0.4.3/lib/asciimath2unitsml/parse.rb:124:in `MathML2UnitsML'
	from /Users/me/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/bundler/gems/metanorma-standoc-5493fa9a68b6/lib/metanorma/standoc/cleanup_maths.rb:215:in `block in mathml_cleanup'
```

**Comment by opoudjis** [comment]
Indeed:

````
      exponent = /\^\(-?\d+\)/.r.map { |m| m.sub(/\^/, "").gsub(/[()]/, "") } |
        /\^-?\d+/.r.map { |m| m.sub(/\^/, "") }
````

.... but I seriously have to question why we would make `kg*m^(p)*s^(q)` a UnitsML expression.

What is the point of UnitsML? To create references to well-defined, semantically understood units of measurement, and hyperlink to their semantic definitions. `kg*m(2)*s^(-1)` is a well-defined, semantically understood unit of measurement. It has a known number of dimensions, it has a defined associated quantity.

...  `kg*m^(p)*s^(q)` is not a unit. It is a unit template. You won't be able to map  `kg*m^(p)*s^(q)` to any quantities, any abbreviations, any known pre-existing  units. Any UnitsML expression it generates will be unusable in the ways we expect UnitsML to be used.

I believe that `unitsml(kg*m^(p)*s^(q))` is a category error, and non-numeric exponents should not be supported by asciimath2unitsml.

**Comment by ronaldtse** [comment]
@opoudjis I agree that this is exactly why it should fail, because this statement cannot be resolved into a real unit.

However, the failure is not explicit in explaining this. Instead of a parsing error, we should expose an error that the unit cannot be defined because the dimension and quantity are not provided.

**Comment by opoudjis** [comment]
Ah! ok, cool. But... parser. It's a parser error. Explaining that will be tricky...

**Comment by opoudjis** [comment]
@suleman-uzair This becomes something you should handle, now that Plurimath encompasses unitsml

At issue: if the exponents of unit expressions in UnitsML are not numbers but variables, e.g. `unitsml(kg*m^(p)*s^(q))` instead of `unitsml(kg*m^(-2)*s^(-2))`, an error needs to be raised, to indicate that this is not a real unit, and is thus not legal to pass into unitsml.

**Comment by suleman-uzair** [comment]
@opoudjis
Sure, I'll check it after my current task. If it's urgent(high priority), let me know, and I'll prioritize it.

**Comment by opoudjis** [comment]
No, not urgent, it's been sitting unactioned for months. Just don't want it to be even more months :)

**Comment by suleman-uzair** [comment]
@opoudjis
Closing this issue since it's fixed in the [PR#209](https://github.com/plurimath/plurimath/issues/209).
